src/main/java/core/basesyntax/dao/machine/MachineDaoImpl.java

```diff
@@ -12,11 +15,34 @@ protected MachineDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Machine save(Machine machine) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(machine);
+            transaction.commit();
+            return machine;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Couldn't insert machine " + machine, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Machine> findByAgeOlderThan(int age) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            int year = LocalDate.now().getYear() - age;
+            return session.createQuery("FROM Machine "
+                    + "WHERE year > :year", Machine.class)
```

Comment:
Older than, means that production year is lower than year in parameter. In your case year is greater.

Reply:
Right